Add gel explosion tower

Fenway17 opened the pull request:
Awaiting assets for gel tower

Fenway17 committed:
Added skeleton for gel explosion tower

Fenway17 committed:
Added subclass for bullet for gel explosion tower

- currently the slow will conflict with freezing, require changes to enemy script

Fenway17 committed:
Fixed bugs in code

Fenway17 committed:
Implemented extra effect manager to handle gel slow and future effects not tied to elements

- updated enemy script to reflect this
- updated gel tower and bullet scripts to reflect this
- added extra effect manager script to ALL enemy prefabs

Fenway17 committed:
Created gel tower assets using dummy cannon assets

trxe commented on Assets/Scripts/ExtraEffectSystem/ExtraEffectManager.cs:
```diff
@@ -0,0 +1,61 @@
+ * This manages extra effects on enemies not tied to elements, and it applied 
+ * after element effects
+ */
+public class ExtraEffectManager : MonoBehaviour {
```
can remove this already

trxe committed:
EnemyEffectManager

Fenway17 committed:
Merge pull request #223 from CS3247-Game-Development-Team-6/add-gel-tower-effect-manager

Suggested changes to ExtraEffectManager --> EnemyEffectManager

trxe commented on Assets/Scripts/Enemy/Enemy.cs:
```diff
@@ -76,6 +76,11 @@ public class Enemy : MonoBehaviour {
+    /**
+     * ExtraEffectManager
+     */
+    private ExtraEffectManager extraEffectManager;
```
Once EnemyEffectManager is ok already, remove this

trxe commented on Assets/Scripts/ExtraEffectSystem/Effect.cs:
```diff
@@ -0,0 +1,6 @@
+using System.Collections;
+public interface Effect {
```
help rename to EnemyEffect thx

trxe reviewed (changes requested):
please edit by tonight, may need some of these classes. thank you

Fenway17 committed:
Reimplemented enemy active effect system

- Reimplemented gel effect
- Changed enemy prefab to reflect new manager for the effects

Fenway17 committed:
shifted variables around

trxe committed:
creates a new multiplier for the enemy speed to fix bug

trxe committed:
rename EnemyActiveEffectsManager to EnemyActiveEffects

trxe committed:
add towers, haven't added bullet prefabs per tower

Fenway17 committed:
Merge pull request #224 from CS3247-Game-Development-Team-6/add-gel-tower-fix

[Gel Tower] Fix speed bug and cleanup+generalize

Fenway17 committed:
Fixed asset for bullet prefab

Fenway17 committed:
Fixed bullet prefab assignment

Fenway17 committed:
Added stats to gel explosion tower scriptable objects
